apps/frontend/src/components/filters.tsx

```diff
@@ -0,0 +1,215 @@
+import {
+  Grid,
+  Button,
+  Box,
+  FormControl,
+  OutlinedInput,
+  InputAdornment,
+  MenuItem,
+  SelectChangeEvent,
+  Select,
+  Typography,
+  FormHelperText,
+  useMediaQuery,
+} from "@mui/material";
+import { useStyles } from "../styles/pageStyles";
+import * as React from "react";
+import { colors } from "../styles/colors";
+import { ChangeEvent } from "react";
+
+interface SearchObject {
+  query: string;
+  minPrice: number;
+  maxPrice: number;
+  status: string;
+  searchType: string;
+  latitude: number;
+  longitude: number;
+  sort: string;
+  page: number;
+  limit: number;
+}
+
+interface FiltersProps {
+  onFilterChange: (filters: Partial<SearchObject>) => void;
+}
+
+const Filters: React.FC<FiltersProps> = ({ onFilterChange }) => {
+  const classes = useStyles();
+
+  const [minPrice, setMinPrice] = React.useState<number | "">("");
+  const [maxPrice, setMaxPrice] = React.useState<number | "">("");
+  const [status, setStatus] = React.useState<string>("");
+  const [type, setType] = React.useState<string>("LISTING");
+
+  const handleMinPriceChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setMinPrice(
+      event.target.value === "" ? "" : parseFloat(event.target.value),
+    );
+  };
+
+  const handleMaxPriceChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setMaxPrice(
+      event.target.value === "" ? "" : parseFloat(event.target.value),
+    );
+  };
+
+  const handleStatusChange = (event: SelectChangeEvent<string>) => {
+    setStatus(event.target.value);
+  };
+
+  const handleTypeChange = (event: SelectChangeEvent<string>) => {
+    setType(event.target.value);
+  };
+
+  const handleApplyFilters = () => {
+    // Pass the current filter values to the parent component only when "Apply Filters" button is clicked
+    onFilterChange({
+      minPrice: minPrice === "" ? undefined : minPrice,
+      maxPrice: maxPrice === "" ? undefined : maxPrice,
+      status,
+      searchType: type,
+    });
+  };
+
+  const handleClearFilters = () => {
+    setMinPrice("");
+    setMaxPrice("");
+    setStatus("");
+    setType("LISTING");
+    onFilterChange({
+      minPrice: undefined,
+      maxPrice: undefined,
+      status: "",
+      searchType: "",
+    });
+  };
+
+  const isDesktop = useMediaQuery("(min-width:850px)");
```

Comment:
I believe in the theme there are values defined for sm, md, lg and xl for screen sizes. These should be used for consistency if the theme variables are updated

Reply:
I did not update this because I needed the page size that was checked to be specific